src/django/api/views.py

```diff
@@ -1057,14 +1057,86 @@ def split(self, request, pk=None):
         except Facility.DoesNotExist:
             raise NotFound()
 
-    @action(detail=True, methods=['GET', 'POST'],
+    @action(detail=True, methods=['POST'],
             permission_classes=(IsRegisteredAndConfirmed,))
     @transaction.atomic
     def promote(self, request, pk=None):
         if not request.user.is_superuser:
             raise PermissionDenied()
+        match_id = request.data.get('match_id')
+
+        if match_id is None:
+            raise BadRequestException('Missing required param match_id')
+
+        try:
+            facility = Facility \
+                .objects \
+                .get(pk=pk)
+
+            match = FacilityMatch \
+                .objects \
+                .get(pk=match_id)
+
+            matched_statuses = [
+                FacilityListItem.MATCHED,
+                FacilityListItem.CONFIRMED_MATCH,
+            ]
+
+            if match.facility_list_item.status not in matched_statuses:
+                raise BadRequestException('Incorrect list item status')
+
+            previous_created_from_id = facility.created_from.id
 
-        return Response('ok')
+            facility.name = match.facility_list_item.name
+            facility.address = match.facility_list_item.address
+            facility.country_code = match.facility_list_item.country_code
+            facility.location = match.facility_list_item.geocoded_point
+            facility.created_from = match.facility_list_item
+            facility.save()
```

Comment:
Consider setting `facility.changeReason` before saving to something like "Promoted list item {new list item id} from {new list as string} over list item {previous list item id} from {previous list as string}"

Reply:
I added this in 5b5ce2bee073396f3fc9875aabadc7c59420dd36